ambry-rest/src/main/java/com/github/ambry/rest/PublicAccessLogger.java

```diff
@@ -13,27 +13,33 @@
  */
 package com.github.ambry.rest;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
+//import org.slf4j.Logger;
+//import org.slf4j.LoggerFactory;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.message.StringMapMessage;
 
 /**
  * Logs requests and responses to public access log.
  */
 public class PublicAccessLogger {
 
-  private static final Logger publicAccessLogger = LoggerFactory.getLogger("PublicAccessLogger");
+  //private static final Logger publicAccessLogger = LoggerFactory.getLogger("PublicAccessLogger");
+  private static final Logger publicAccessLogger = LogManager.getLogger("PublicAccessLogger");
```

Comment:
Just wanted to note that this seems to directly use the Log4j APIs instead of SLF4J (which seems to be an [abstraction](https://howtodoinjava.com/log4j2/slf4j-vs-log4j-which-one-is-better/) over Log4j and can be useful for replacing with other logging frameworks). For example, we provide the log4j-slf4j-impl in the dependencies [here](https://github.com/linkedin/ambry/blob/ca77dc6bec8a84c3f2200295eba6688422f2adcf/build.gradle#L430).

But, I couldn't find on how to use structured logging (StringMapMessage) with SLF4J. May be it is okay for now?

Reply:
According to the main Wiki page that I've used for guidance at https://iwww.corp.linkedin.com/wiki/cf/pages/viewpage.action?spaceKey=ENGS&title=Structural+Logging+with+Log4J we're not yet able to use SLF4J for structured logging at LinkedIn:

"Specific for structural logging, it is not supported until SLF4j version 2.x, which is not allowed yet in LinkedIn due to the compatibility issues. Therefore we now recommend Log4j API for structural logging. SLF4j support will be added later on."